var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -105,6 +105,10 @@ class Qt(Package):
           working_dir='qtbase',
           when='@5.10:5.12.0 %gcc@9:')
 
+    # Fix build of QT4 with GCC 9
+    # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=925811
+    patch("qt4-gcc9-qforeach.patch", when="@4: %gcc@9")
```

Comment:
This patch works for Qt 5 too? If not, you'll need to change it to something like:
```suggestion
    patch("qt4-gcc9-qforeach.patch", when="@4.0:4.999 %gcc@9")
```